bundle/src/main/java/com/adobe/acs/commons/packaging/impl/AssetPackagerServletImpl.java

```diff
@@ -0,0 +1,281 @@
+                if (excludedPages.size() > 0) {
+                    for (String excludedPath : excludedPages) {
+                        pageFilter.addExclude(new DefaultPathFilter(excludedPath));
+                    }
+                }
+                paths.add(0, pageFilter);
+            }
+
+            doPackaging(request, response, preview, properties, paths);
+        } catch (RepositoryException ex) {
+            log.error("Repository error while creating Asset Package", ex);
+            response.getWriter().print(packageHelper.getErrorJSON(ex.getMessage()));
+        } catch (IOException ex) {
+            log.error("IO error while creating Asset Package", ex);
+            response.getWriter().print(packageHelper.getErrorJSON(ex.getMessage()));
+        }
+    }
+
+    /**
+     * Gets the properties saved to the Asset Packager Page's jcr:content node.
+     *
+     * @param request The request obj
+     * @return A ValueMap representing the properties
+     */
+    private ValueMap getProperties(final SlingHttpServletRequest request) {
+        if (request.getResource().getChild("configuration") == null) {
+            log.warn("Asset Packager Configuration node could not be found for: {}", request.getResource());
+            return new ValueMapDecorator(new HashMap<>());
+        } else {
+            return request.getResource().getChild("configuration").adaptTo(ValueMap.class);
+        }
+    }
+
+    /**
+     * Takes a list of Strings that should be valid patterns. If they are not then the exception is
+     * logged and the pattern is not added to the list of checked patterns.
+     *
+     * @param input The array of intended patterns
+     * @return The converted array of patterns
+     */
+    private List<Pattern> generatePatterns(String[] input) {
+        final List<Pattern> patterns = new ArrayList<>();
+        for (String item : input) {
+            try {
+                patterns.add(Pattern.compile(item));
+            } catch (PatternSyntaxException e) {
+                log.error("Pattern invalid, skipping. Pattern value: " + item);
+            }
+        }
+        return patterns;
+    }
+
+    /**
+     * Recursively iterate over the parent path specified in the configuration to aggregate all the
+     * String or String Array property values that are referencing a path in the DAM.
+     *
+     * @param resourceResolver Resolver from the current Request
+     * @param pagePath         The page path
+     * @return The full list of filter sets
+     */
+    private List<PathFilterSet> findAssetPaths(final ResourceResolver resourceResolver,
+                                               final String pagePath) {
+
+        final List<PathFilterSet> filters = new ArrayList<>();
+        final Resource parentResource = resourceResolver.resolve(pagePath);
+        if (isExcluded(this.pageExclusionPatterns, pagePath)) {
+            final PageManager pageManager = resourceResolver.adaptTo(PageManager.class);
```

Comment:
Is this logic to get the containing page path instead of the current `pagePath` required for some sort of edge case?  Struggling to think of the case where it's needed.

Reply:
So the `findAssetPaths` method recurses over all the resources underneath a given path, so the `parentResource` defined here can be something like `/content/we-retail/language-masters/en/experience/arctic-surfing-in-lofoten/jcr:content/root/hero_image`. The point of the 'page exclusions' is to exclude content from certain pages, so it's getting the page of each resource that's being iterated over and checking if it's excluded. Essentially the point is to compare the page exclusions against the page currently being looked at (from any level).

Reply:
Correct me if I'm wrong, but if `/content/we-retail/language-masters/en/experience/arctic-surfing-in-lofoten` is already in the exclusion paths then we will never recurse down to `/content/we-retail/language-masters/en/experience/arctic-surfing-in-lofoten/jcr:content/root/hero_image` in the first place, will we?

Reply:
ok, just seemed like we were running extra unnecessary code here, but I dont see any real harm in the way it is written